src/ModelContextProtocol.Core/Client/StdioClientTransport.cs

```diff
@@ -156,20 +156,21 @@ public async Task<ITransport> ConnectAsync(CancellationToken cancellationToken =
             // up the encoding from Console.InputEncoding. As such, when not targeting .NET Core,
             // we temporarily change Console.InputEncoding to no-BOM UTF-8 around the Process.Start
             // call, to ensure it picks up the correct encoding.
-#if NET
-            processStarted = process.Start();
-#else
+
             Encoding originalInputEncoding = Console.InputEncoding;
-            try
+            if (HasConsole())
             {
-                Console.InputEncoding = StreamClientSessionTransport.NoBomUtf8Encoding;
-                processStarted = process.Start();
-            }
-            finally
-            {
-                Console.InputEncoding = originalInputEncoding;
+                try
+                {
+                    Console.InputEncoding = StreamClientSessionTransport.NoBomUtf8Encoding;
+                }
+                finally
+                {
+                    Console.InputEncoding = originalInputEncoding;
+                }
             }
-#endif
+
+            processStarted = process.Start();
```

Comment:
We need start the process before resetting back to the originalInputEncoding for it to matter.